File: fl_camera/ios/Classes/FlCameraEvent.swift
import Flutter
import Foundation

public class FlCameraEvent: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    var eventChannel: FlutterEventChannel?

    init(_ messenger: FlutterBinaryMessenger) {
        super.init()
        eventChannel = FlutterEventChannel(name: "fl.camera.event", binaryMessenger: messenger)
        eventChannel!.setStreamHandler(self)
    }

    public func sendEvent(_ arguments: Any?) {
        eventSink?(arguments)
    }

    func dispose() {
        eventSink = nil
        eventChannel?.setStreamHandler(nil)
        eventChannel = nil
    }

    public func onListen(withArguments arguments: Any?, eventSink event: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = event
        return nil
    }

    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        dispose()
        return nil
    }
}

File: fl_camera/android/src/main/kotlin/fl/camera/FlCameraEvent.kt
package fl.camera

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class FlCameraEvent(binaryMessenger: BinaryMessenger) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var eventChannel: EventChannel? = null
    private var lastArgs: Any? = null

    init {
        eventChannel = EventChannel(binaryMessenger, "fl.camera.event")
        eventChannel!!.setStreamHandler(this)
    }

    fun sendEvent(arguments: Any?) {
        if (lastArgs == arguments) return
        lastArgs = arguments
        eventSink?.success(arguments)
    }

    fun dispose() {
        eventSink?.endOfStream()
        eventSink = null
        eventChannel?.setStreamHandler(null)
        eventChannel = null
    }

    override fun onListen(arguments: Any?, event: EventChannel.EventSink?) {
        eventSink = event
    }

    override fun onCancel(arguments: Any?) {
        dispose()
    }
}